maker_dao/src/agent.ts

```diff
@@ -0,0 +1,163 @@
+              description: `Deposit occur on DAI ${chainId} chainId bridge`,
+              alertId: `${
+                chainId === OP_CHAINID ? "OPTIMISM-1" : "ARBITRUM-1"
+              }`,
+              severity: FindingSeverity.Low,
+              type: FindingType.Info,
+              metadata: {
+                totalSupply: data.toString(),
+                amount: amount.toString(),
+                from: from,
+                to: to,
+              },
+            })
+          );
+        }
+      }
+    }
+
+    if (chainId === ETHER_CHAINID) {
+      if (transferEvent[0]) {
+        const toAddress = transferEvent[0].args?.to?.toLowerCase();
+
+        const { alerts } = await getL1Alert(query);
+
+        for (const transfer of transferEvent) {
+          if (alerts) {
+            const totalSupply = await alerts[0].metadata.totalSupply;
+
+            const blockTag = await txEvent.blockNumber;
+            let Escrowbalance;
+            let a;
+            if (toAddress === L1_AARBITRUM_GATEWAY.toLowerCase()) {
+              Escrowbalance = await GET_DAI_BALANCE(
+                provider,
+                L1_ESCROW_ADDRESS_ARBITRUM,
+                blockTag
+              );
+            } else if (toAddress === L1_ESCROW_ADDRESS_OPTIMISM.toLowerCase()) {
+              Escrowbalance = await GET_DAI_BALANCE(
+                provider,
+                L1_ESCROW_ADDRESS_OPTIMISM,
+                blockTag
+              );
+            } else {
+              return findings;
+            }
+
+            if (Escrowbalance >= totalSupply) {
+              return findings;
+            } else {
+              findings.push(
+                Finding.fromObject({
+                  name: `Scam Transaction Detected on ${
+                    toAddress === L1_AARBITRUM_GATEWAY.toLowerCase()
+                      ? "Arbitrium DAI"
+                      : "Optimism DAI"
+                  }`,
+                  description: `Scam transaction occur on ${
+                    toAddress === L1_AARBITRUM_GATEWAY.toLowerCase()
+                      ? "Arbitrium"
+                      : "Optimism"
+                  } DAI Address. Total supply of ${totalSupply} is greater than Escrow balance of ${Escrowbalance}`,
+                  alertId: `${
+                    toAddress === L1_AARBITRUM_GATEWAY.toLowerCase()
+                      ? "ARBITRUM-2"
+                      : "OPTIMISM-2"
+                  }`,
+                  severity: FindingSeverity.High,
+                  type: FindingType.Suspicious,
+                  metadata: {
+                    totalSupply: totalSupply.toString(),
+                    balance: Escrowbalance.toString(),
+                  },
+                })
+              );
+            }
+          }
+        }
+      }
+    }
+
+    return findings;
+  };
+
+export default {
+  handleTransaction: provideHandleTransaction(
+    PROVIDER,
+    L2_DAI_GATEWAY_ARB,
+    emptyAlertResponse as any
```

Comment:
You are right in thinking that you would need to have a way to "get" some "alerts" (hint: look in the [Forta SDK](https://docs.forta.network/en/latest/sdk/) for method that can do just that. And you're correct that it's type would be `(alertQuery: AlertQueryOptions) => Promise<AlertsResponse>`.

However, as it currently is, the usage of the `emptyAlertResponse` as you have it is not the appropriate approach.

So you would need to use the actual method I am hinting at from the SDK to _"get"_ the _"alerts"_ you need.

Reply:
Won't be able to test it with the get alert method

Reply:
Okay, seeing as how this argument you're passing, `emptyAlertResponse`, won't work in production like you hope, I suggest not passing it as an argument and instead implement a function that would call `getAlerts` inside of its logic. Then you comment out the `getAlerts` call for testing locally and have the logic return the local data you need.

Yash seems to have had a similar discussion with Vasilis as a work around for the breaking change to `getAlerts`, so he if you can sync with him on how he implemented it.